python/simulation.py

```diff
@@ -3169,3 +3169,52 @@ def quiet(quietval=True):
 
 def verbosity(verbose_val):
     mp.cvar.verbosity = verbose_val
+
+def merge_subgroup_data(data,num_groups):
+    # Lazy import
+    from mpi4py import MPI    
+    
+    # Pull required parameters
+    comm = MPI.COMM_WORLD
+    num_workers = comm.Get_size()
+    shape=data.shape
+    size = data.size
+    out_shape = shape + (num_groups,)
+    dtype=data.dtype.char
+    
+    # Initialize new input and output datasets
+    input=np.zeros(shape,dtype,order='F')
+    input[:]=data
```

Comment:
Seems like it would be better to do
```py
input = np.array(data, order='F')
```
and then compute `shape` and `size` from `input`.  That way it will work even if `data` is not a numpy array (e.g. for Python lists).

Reply:
Agreed.